File: Scripts/results/generated_code_results/Flutter/bak_code/Entertainment/ShortVideoApp/CommentsScreen.dart
import 'package:flutter/material.dart';
class CommentsScreen extends StatefulWidget {
@override
_CommentsScreenState createState() => _CommentsScreenState();
}
class _CommentsScreenState extends State<CommentsScreen> {
final TextEditingController _commentController = TextEditingController();
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('4231 Comments'),
actions: [
IconButton(
icon: Icon(Icons.edit),
onPressed: () {
// Implement edit functionality
},
),
],
),
body: Column(
children: [
Expanded(
child: ListView(
children: [
_buildCommentItem('Babayaga', 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit. Aenean sit.', '1.2k', 12),
_buildCommentItem('LilKittt', 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit.', '1.8k', 12),
_buildCommentItem('Spiderman', 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit.', '1.8k', 12),
_buildCommentItem('Badli', 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit.', '1.8k', 12),
_buildCommentItem('Tung Tran', 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit.', '1.8k', 12),
],
),
),
_buildAddCommentSection(),
],
),
);
}
Widget _buildCommentItem(String username, String comment, String likes, int replies) {
return Padding(
padding: const EdgeInsets.all(8.0),
child: Row(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
),
SizedBox(width: 8),
Expanded(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(username, style: TextStyle(fontWeight: FontWeight.bold)),
SizedBox(height: 4),
Text(comment),
SizedBox(height: 4),
Row(
children: [
Row(
children: [
IconButton(
icon: Icon(Icons.thumb_up),
onPressed: () {
// Implement like functionality
},
),
Text(likes),
],
),
SizedBox(width: 16),
TextButton(
child: Text('View replies ($replies)'),
onPressed: () {
// Implement view replies functionality
},
),
],
),
],
),
),
],
),
);
}
Widget _buildAddCommentSection() {
return Padding(
padding: const EdgeInsets.all(8.0),
child: Row(
children: [
CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
),
SizedBox(width: 8),
Expanded(
child: TextField(
controller: _commentController,
decoration: InputDecoration(
hintText: 'Add Comment...',
border: OutlineInputBorder(),
),
),
),
IconButton(
icon: Icon(Icons.send),
onPressed: () {
// Implement send comment functionality
String newComment = _commentController.text;
if (newComment.isNotEmpty) {
// Add the new comment to the list
setState(() {
// Update the state with the new comment
});
_commentController.clear();
}
},
),
],
),
);
}
}

File: Scripts/results/generated_code_results/RN_gpt/Entertainment/ShortVideoApp/CommentsScreen.js
import React from 'react';
import { View, Text, Image, Button, TextInput, StyleSheet, FlatList, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

const commentsData = [
  {
    user: 'Babayaga',
    comment: 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit. Aenean sit.',
    likes: '1.2k',
    replies: 12,
  },
  {
    user: 'LilKittt',
    comment: 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit.',
    likes: '1.8k',
    replies: 12,
  },
  {
    user: 'Spiderman',
    comment: 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit.',
    likes: '1.8k',
    replies: 12,
  },
  {
    user: 'Badli',
    comment: 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit.',
    likes: '1.8k',
    replies: 12,
  },
  {
    user: 'Tung Tran',
    comment: 'Lorem ipsum dolor sit amet, elit poo consectetur adipiscing elit.',
    likes: '1.8k',
    replies: 12,
  },
];

const CommentItem = ({ user, comment, likes, replies }) => {
  return (
    <View style={styles.commentContainer}>
      <Image source={require('../assets/snack-icon.png')} style={styles.profileImage} />
      <View style={styles.commentContent}>
        <Text style={styles.username}>{user}</Text>
        <Text style={styles.commentText}>{comment}</Text>
        <View style={styles.actions}>
          <TouchableOpacity>
            <Text style={styles.viewReplies}>View replies ({replies})</Text>
          </TouchableOpacity>
          <View style={styles.likeButton}>
            <Icon name="heart" size={20} color="red" />
            <Text style={styles.likeText}>{likes}</Text>
          </View>
        </View>
      </View>
    </View>
  );
};

const CommentsScreen = () => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headerText}>4231 Comments</Text>
        <TouchableOpacity>
          <Icon name="edit" size={24} color="white" />
        </TouchableOpacity>
      </View>
      <FlatList
        data={commentsData}
        renderItem={({ item }) => (
          <CommentItem user={item.user} comment={item.comment} likes={item.likes} replies={item.replies} />
        )}
        keyExtractor={(item) => item.user}
      />
      <View style={styles.addCommentContainer}>
        <TextInput style={styles.commentInput} placeholder="Add Comment..." placeholderTextColor="#888" />
        <TouchableOpacity>
          <Icon name="send" size={24} color="white" />
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'black',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 16,
    borderBottomWidth: 1,
    borderBottomColor: '#333',
  },
  headerText: {
    fontSize: 18,
    color: 'white',
  },
  commentContainer: {
    flexDirection: 'row',
    padding: 16,
    borderBottomWidth: 1,
    borderBottomColor: '#333',
  },
  profileImage: {
    width: 50,
    height: 50,
    borderRadius: 25,
    marginRight: 12,
  },
  commentContent: {
    flex: 1,
  },
  username: {
    fontSize: 16,
    fontWeight: 'bold',
    color: 'white',
  },
  commentText: {
    fontSize: 14,
    color: 'white',
    marginVertical: 4,
  },
  actions: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  viewReplies: {
    color: '#3b82f6',
  },
  likeButton: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  likeText: {
    marginLeft: 4,
    color: 'white',
  },
  addCommentContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 16,
    borderTopWidth: 1,
    borderTopColor: '#333',
  },
  commentInput: {
    flex: 1,
    padding: 10,
    backgroundColor: '#222',
    borderRadius: 20,
    marginRight: 8,
    color: 'white',
  },
});

export default CommentsScreen;